### PR TITLE
Migrate .clearfix

### PR DESCRIPTION
**Pull request by WiNloSt**
Epic #38811
Sub-task #41239

### Description

Migrate `clearfix.modules.css`

### How to verify

One of the examples is to go to admin > troubleshooting and observe the page header.

If that looks good then I think the rest will be fine, since the migration is pretty straightforward.

### Demo
Screenshot 2024-04-17 at 5 34 33 PM

### Checklist

- [ ] ~Tests have been added/updated to cover changes in this PR~ Only style change

### CODE REVIEW

File: frontend/src/metabase/css/core/clearfix.module.css [hunk 1]
```diff
@@ -10,20 +10,12 @@
...
-:global(.clearfix::after) {
+.clearfix::after {
   clear: both;
 }
-
-/**
- * For IE 6/7 only
- * Include this rule to trigger hasLayout and contain floats.
- */
-:global(.clearfix) {
```
Comment (WiNloSt) [inline frontend/src/metabase/css/core/clearfix.module.css:27]: We even ended support for IE11, so I think this rule is pretty safe to be removed.

**Review by heypoom — APPROVED** [review]
I double-checked for references to `clearfix` and indeed there is only 3 of them. Looks good!

**Comment by uladzimirdev** [comment]
@WiNloSt couldn't we just drop it? 😄 I didn't use clearfix for more than 5 years and it's useless in the world of flexbox

**Comment by WiNloSt** [comment]
> @WiNloSt couldn't we just drop it? 😄 I didn't use clearfix for more than 5 years and it's useless in the world of flexbox

I was thinking about that, but just to accelerate migration CSS, I didn't bother trying to change some of it to flex. There are only some usages from this PR and [this float PR](https://github.com/metabase/metabase/pull/41510/files).

**Pull request merged**